backend/src/project/endpoints.ts

```diff
@@ -31,6 +47,19 @@ export const getProjectById = async (req: Request, res: Response) => {
                 res.status(404).json({ error: "No project found with given id" });
                 return;
             } else {
+                const loggedUser = isAuthenticated(req);
+
+                if (!loggedUser) {
```

Comment:
To sprawdzanie czy user ma dostęp do danego projektu przeniósłbym do nowej funkcji w project/service.ts, która przyjmuje 2 argumenty: `project`, `user` i zwraca `bool`. I ten warunek, które napisałeś da się uprościć:
`project.author === "AlgoDebug || user && project.author === user"` (zrobiłem negację całości, żeby zwróciło true, jeżeli user ma dostęp)